desktop-widgets/tab-widgets/TabDiveInformation.cpp

```diff
@@ -94,9 +93,10 @@ void TabDiveInformation::closeWarning()
 }
 
 void TabDiveInformation::updateWaterTypeWidget()
-{    // Decide on whether to show the water type/salinity combobox or not
-	if (prefs.salinityEditDefault || manualDive)
-	{         // if the preference setting has been checked or this is a manually-entered dive
+{
+	// Decide on whether to show the water type/salinity combobox or not
+	bool hasDCSalinity = parent.currentDive->salinity != 0;
```

Comment:
@rafsalvioni: This is causing SEGV crashes on startup in some circumstances - looks like `currentDive` needs to be null checked here.

Reply:
@mikeller I fixed in https://github.com/rafsalvioni/subsurface/commit/865a37de90a06f0c6c4e8b83abfcfb849e2abc35

Are you going to reopen PR or I need to create a new one?

Reply:
I don't think there is a way to re-open a pull request once it's been merged, so please open a new one.